WordPress/gutenberg
# Blocks: update warning style

## Description
Update the block warning style, moving the action buttons inline with the text, and adding a grey scrim. This prepares the warning component for the addition of a dropdown options menu. Discussion and reasons behind these changes can be found in #7604

![vlidate](https://user-images.githubusercontent.com/1277682/42449279-f06a048e-8377-11e8-8074-f57e98e4d979.jpg)

The block warning is shown in two situations:
- A block crashes and a `<BlockCrashWarning />` component is shown
- A block contains unexpected content and a `<BlockInvalidWarning />` is shown

In addition to style changes some text has been changed:
- The `Edit as HTML` button is now `Keep as HTML` to better reflect the action
- The message `This block appears to have been modified externally.` is now `This block has been modified externally.` to remove general vagueness

These items of text are likely to change in the future as the validation experience is improved, but for now should help with confusion.

Also to note that `InvalidBlockWarning` has been renamed to `BlockInvalidWarning` so it matches the other block components.

## How has this been tested?
Existing unit tests cover the changes to the HTML.

To manually test the block validation message:
- Create a paragraph
- Edit as HTML and remove the trailing `</p>` to trigger the block validation message
- Verify that the message changes from:
![another_post___ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/42451600-9cf25880-837f-11e8-847e-973c227e92d3.jpg)
to:
![vlidate](https://user-images.githubusercontent.com/1277682/42449279-f06a048e-8377-11e8-8074-f57e98e4d979.jpg)
- The buttons are right aligned and a grey scrim overlays the block content
- Verify that as the page width decreases the warning adjusts the buttons beneath the text (still left aligned):
![another_post___ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/42510867-27fadd1c-8448-11e8-9c8f-768bdb8c6494.jpg)
- Verify that the buttons continue to work as before

To test the block crash warning message:
- From a developer console type `wp.blocks.registerBlockType( 'myplugin/mr-crashy', { title: 'Mr. Crashy', category: 'common', icon: 'warning', edit() { throw new Error(); }, save() { return ''; } } );`
- Add `Mr Crashy` block to a post
- Confirm that the crash warning has changed from:
![another_post___ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/42457586-bec68a02-838f-11e8-8f07-694ecd2088f5.jpg)

To:

![another_post___ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/42457439-6470a68c-838f-11e8-9cf8-3b6a4f49f695.jpg)
- The warning now matches the width of a block

## Types of changes
This is a non-breaking change. The block warning functionality doesn't change.

## Checklist:
